internal/clients/aws.go

```diff
@@ -111,6 +113,37 @@ func (m *metaOnlyPrimary) Meta() any {
 	return m.meta
 }
 
+// withExternalAPICallCounter configures an AWS SDK v2 stack (client)
+// with an API call counter.
+func withExternalAPICallCounter(stack *middleware.Stack) error {
+	externalAPICallCounterMiddleware := middleware.DeserializeMiddlewareFunc("externalAPICallCounter",
+		func(ctx context.Context, input middleware.DeserializeInput, next middleware.DeserializeHandler) (middleware.DeserializeOutput, middleware.Metadata, error) {
+			serviceID := awsmiddleware.GetServiceID(ctx)
+			operationName := awsmiddleware.GetOperationName(ctx)
+
+			output, metadata, err := next.HandleDeserialize(ctx, input)
```

Comment:
Could you please explain explicitly in a comment that `next.HandleDeserialize` actually invokes the rest of the chain including the round-tripper and reports back the result?